src/Kunstmaan/AdminListBundle/Helper/DoctrineDBALAdapter.php

```diff
@@ -66,10 +71,19 @@ public function getNbResults()
         if ($this->useDistinct) {
             $distinctString = 'DISTINCT ';
         }
-        $statement = $query->select('COUNT(' . $distinctString . $this->countField . ') AS total_results')
-            ->orderBy($this->countField)
-            ->setMaxResults(1)
-            ->execute();
+
+        if ($this->databasePlatform == 'postgresql') {
```

Comment:
Is this check necessary? I saw that in [#2811](https://github.com/Kunstmaan/KunstmaanBundlesCMS/pull/2811/files#diff-ad01d84409afe7ac963af2ce17fd3a076a100eed9c35b6d5e13c69cc75ef46df) the order by was removed and that seems like the right solution as a count with order by isn't needed I think

Reply:
Yes, I believe this check to be neccesary. In #2811 orderby was removed becaused it creates postgres errors when used in conjunction with COUNT as it is here. Simply removing the order by - because its not neccesary- from the original kunstmaan code is unfortunately not enogh, this may work for simple queries that are provided by querybuilder. However when queries provided by queryBuilder get more complex and contain orderBys in themselves we are back to square one. Hence the check

Reply:
Sometimes the queryBuilder query itself may contain orderBy clauses which will reek havoc in postgress when used in conjuction with COUNT DISTINCT. Therefore it is not sufficient to simply remove ones Kunstmaans own orderBy clause, but also the orderby clauses of the querybuilder query itself.  IMO the reason that the orderBy was removed from the kunstmaan code was not that it is not neccesary, but in postgres it is detrimental.

Reply:
Yes I know postgres has issues with that, but what I meant was can't this be done for "all" (mysql and postgress) platforms? So we remove the general orderBy and reset the OrderBy on the query object.

Reply:
OK, made the change also for non postgress and extensively retested that against mysql database. You are right. This works also for mysql, so no if statement needed and cleaner code. Thanks!